### DISCUSSION TITLE
Title generation failing with Pipelines

### DISCUSSION THREAD

**Discussion by DarkIceXD** [danny-avila/LibreChat#4751] [Troubleshooting] [answered]
### What happened?

Hello, i am trying to use [pipelines](https://github.com/open-webui/pipelines) within of Librechat and the chatting functionality works but title generation doesnt.

I found that the error is coming from [here](https://github.com/danny-avila/LibreChat/blob/main/api/app/clients/OpenAIClient.js#L1428-L1441).

### Steps to Reproduce

1. Set up pipelines using docker.
2. Install a basic pipeline e.g. return a string.
3. Add this to your librechat.yaml

> - name: "Pipelines"
>       apiKey: "0p3n-w3bu!"
>       baseURL: "http://pipelines:9099/v1/chat/completions"
>       models:
>         default: ["foo"]
>       titleConvo: true
>       titleModel: "foo"
>       titleMessageRole: "user"
>       summarize: false
>       summaryModel: "current_model"
>       forcePrompt: false
>       modelDisplayLabel: "Pipelines"

4. send any question through librechat

### What browsers are you seeing the problem on?

Firefox

### Relevant log output

```shell
warn: [OpenAIClient.chatCompletion][finalChatCompletion] Aborted Message
warn: [OpenAIClient.chatCompletion][finalChatCompletion] API error
warn: [OpenAIClient.chatCompletion][stream] Unhandled error type
warn: [OpenAIClient.chatCompletion][finalChatCompletion] Unhandled error type
error: [OpenAIClient] Known OpenAI error: Error: missing role for choice 0
warn: [OpenAIClient] Chat completion response has no choices
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
This looks like an issue with “pipelines” and not LibreChat, seems it’s not fully OpenAI compatible

**Reply by DarkIceXD** [reply]
thanks. you are right. i am going to open a PR in the pipelines repo.
